src/main/java/com/sombra/edu/springmenteeproject/service/impl/TransferServiceImpl.java

```
@@ -0,0 +1,90 @@
+package com.sombra.edu.springmenteeproject.service.impl;
+
+import com.sombra.edu.springmenteeproject.dto.MoneyTransferDTO;
+import com.sombra.edu.springmenteeproject.entity.Status;
+import com.sombra.edu.springmenteeproject.entity.Transfer;
+import com.sombra.edu.springmenteeproject.entity.TransferRequest;
+import com.sombra.edu.springmenteeproject.repository.TransferRepository;
+import com.sombra.edu.springmenteeproject.service.*;
+import jakarta.transaction.Transactional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+
+@Service
+@Slf4j
+public class TransferServiceImpl implements TransferService {
+
+    private final TransferRepository transferRepository;
+    private final UserAccountService userAccountService;
+    private final WalletService walletService;
+    private final BalanceService balanceService;
+    private final TransferRequestService transferRequestService;
+
+    @Autowired
+    public TransferServiceImpl(TransferRepository transferRepository, UserAccountService userAccountService, WalletService walletService, BalanceService balanceService, TransferRequestService transferRequestService) {
+        this.transferRepository = transferRepository;
+        this.userAccountService = userAccountService;
+        this.walletService = walletService;
+        this.balanceService = balanceService;
+        this.transferRequestService = transferRequestService;
+    }
+
+    @Transactional
+    @Override
+    public Transfer sendMoneyToAnotherUser(MoneyTransferDTO transferDTO) {
+        TransferRequest transferRequest = getTransferRequest(transferDTO);
+        if (!isValidTransfer(transferDTO)) {
+            return getTransfer(transferRequest, Status.FAILED);
```

If validation doesn't pass throw an error instead of returning failed transfer